### PR TITLE
Try using left borders for hover + selection states

### PR DESCRIPTION
**Pull request by kjellr**
This PR is a possible way to address two problems: 

1. Our current blue hover state is too prominent. (https://github.com/WordPress/gutenberg/issues/14095)
2. Our current block selection state is too low contrast. (https://github.com/WordPress/gutenberg/issues/12254)

For visual reference, here's how our hover and selection states currently work:

hover-focus-progression

This PR removes the blue outline from our hover state, and replaces it with a blue left border. It adjusts the location of the block breadcrumb to match. The PR also adds a dark left border to the block and block toolbar, to improve accessibility there: 

hover-focus-progression

## Screenshots

Here's a GIF of this in action: 

hover-select

And an example on mobile:
_(there are no hover states on mobile, but the left border is now present here)_

screen shot 2019-02-27 at 10 44 38 am

---

Some considerations: 

- In general, this makes the whole interface feel a little heavy on the left side. This might be fine, but it was a little weird to get used to.
- When blocks are set to `alignfull`, the left border only appears on the block toolbar (unless you're on mobile, in which case it does not appear at all in this case).   
- We can likely do some iteration on the thickness of the left border. I chose `3px` to start with.
- As per @afercia's clarification in https://github.com/WordPress/gutenberg/issues/14095#issuecomment-467192854, the blue color in our current hover state was not chosen due to accessibility concerns. Since we may have some flexibility on that color, I suggest we try a slightly lighter gray color for the hover indicator instead. 
- As raised by @LukePettway in https://github.com/WordPress/gutenberg/pull/13700#issuecomment-466514788, this hover treatment does not outline the entire interactive area. It also may be lost on very large screens. We should try this out and assess the impact of this. 
- When scrolling, the left border of the block can overlap with that of the block toolbar. Since they're using opacity, [the overlap is visible if you look closely](https://cloudup.com/cEPGKH0VJ0q). We may want to try preventing that.

### CODE REVIEW

**Comment by kjellr** [comment]
> ... I suggest we try a slightly lighter gray color for the hover indicator instead.

For reference, here's how this PR looks with `$light-gray-500` hovers, instead of the blue: 

hover-select-gray-single

---

hover-select-gray

---

It feels nice and light to me, negating a bit of the left-heaviness I experienced in the original PR. It does make the hover less prominent, but that may be workable.

**Comment by chrisvanpatten** [comment]
As you noted on the other ticket it is super hard to crawl through the history of this — I was even involved in some of these conversations and don't remember exactly where/when they happened 😂

However, I'm like 80% sure there was a reason that the block name label was moved to the top right, instead of top left. I think it was blocking interaction with the previous block, or maybe causing issues with the adjacent inserter?

cc @jasmussen or @ZebulanStanphill who I _think_ were also in those issues, and might remember better! 🧠

**Comment by mapk** [comment]
I really appreciate the hover states not interfering with my view of the document as a whole. The heaviness on the left felt fine to me. 

Initially, I thought it might look odd with the quote block, but it seemed fine.

screen shot 2019-02-27 at 8 29 35 am

I did notice some rough parts around the columns block, but what isn't rough there right now?

screen shot 2019-02-27 at 8 34 34 am

> For reference, here's how this PR looks with `$light-gray-500` hovers, instead of the blue:

Did you happen to test the a11y on the block label? Does the black on grey contrast work?

**Comment by kjellr** [comment]
> I did notice some rough parts around the columns block, but what isn't rough there right now?

Ah, good catch. Yeah, I need to adjust this for that block. 

> Did you happen to test the a11y on the block label? Does the black on grey contrast work?

Yep, the contrast there has a pretty high `13.17:1` ratio. That's actually a better ratio than what we have against the blue.

**Comment by ZebulanStanphill** [comment]
@chrisvanpatten I don't recall anything specific for why the hover label was on the right, but it is important to note that in this PR, the label is shown _above_ the block border, rather than within it. As long as the label doesn't remain on-screen when you hover over it, it shouldn't cause any issues with selecting the block above.

**Comment by jasmussen** [comment]
> However, I'm like 80% sure there was a reason that the block name label was moved to the top right, instead of top left. I think it was blocking interaction with the previous block, or maybe causing issues with the adjacent inserter?

The very first hover labels were horrendous. I can say that because I designed them, so I'm only offending myself. They were all sorts of in the way, covering sibling inserters and, yeah, it really helped making them small. 

I believe we moved them to the right with the intent of making them into the block drag handle. But since then, obviously, the actual drag handle has landed. On the left. So I think it's okay to explore moving that label around.

---

@kjellr Thank you so very much for exploring this, _head on_. Building a block editor has definitely been a challenge, and although it's thrilling to see a thousand blocks _actually_ blooming (this was always our greatest hope), along the way it's been incredibly arduous to try and find the right balance between being an editor, and being a block editor. I love that we are continuing to explore that, it can only improve things.

In this case, I also deeply love how you state at the top what problems you are trying to solve with this: the heaviness of the UI, and at the same time the lack of contrast with the selected block.

Having just played around with this a little bit in the branch, I feel you _do_ accompolish the contrast issue, but that the heaviness is less successful. Like Mark says, it's nice for the document to feel continuous, but the wider border actually feels heavier to me than the border all the way around. It's hard to put into words, but it almost feels like the document is leaning towards the left now.

But the _ingredients_ seem right. Specifically the left border on hover, sans color.

How would it look if the hover border remained 1px, but was $dark-gray-150 instead, and remained $dark-gray-150 when selected?

**Comment by afercia** [comment]
> and remained $dark-gray-150 when selected?

🤔

**Comment by chrisvanpatten** [comment]
@afercia Is that an "I'm interested to see how this would look" or a "This is probably a bad idea" thinking emoji?


Thanks for the history @jasmussen. I agree it feels a little heavy on the left on select. I like the thick border on hover for sure… less sure about the focus state.

**Comment by afercia** [comment]
It's a thinking emoji. As in: it's basically what the accessibility feedback proposed since the beginning. As long as `$dark-gray-150` gives a contrast of at least 3:1 with the background and it stays `$dark-gray-150` without fading, the thinking emoji thinks it's interesting.

**Comment by folletto** [comment]
FWIW I appreciated the line-on-the-side idea even in the early explorations even if not all the pieces had yet fallen into place, so this might be a good time.

I also feel that this approach with the label on the left looks almost like that the label "expands" into the toolbar, which feels a subtle yet good correlation to me.

I'd also +1 to have the border gray (whichever specific gray is evaluated as appropriate). I agree the blue for hover feels off, as it signals "selected" more than "hover". I vaguely recall why it was a good idea at the time, and again I think now might be the right moment to switch :)

**Comment by kjellr** [comment]
A couple quick notes: 

Based on the feedback in this thread, I've updated this PR to use a $light-gray-500 hover state, [as mocked up above](https://github.com/WordPress/gutenberg/pull/14145#issuecomment-467931958): 

53505809-9f0b7b00-3a82-11e9-8a3d-100e20d15df3

Give the PR another test and please share your impressions. 

---

> How would it look if the hover border remained 1px, but was $dark-gray-150 instead, and remained $dark-gray-150 when selected?

I've opened #14197 with a version of this idea. It uses a `$dark-gray-300` left border (I had to darken it from your suggestion in order to get the breadcrumb to pass AA), alongside `$dark-opacity-200` borders for the focus state. I adjusted the borders so that they surround the block and the block toolbar, but do not effect the dividers inside of the toolbar. Compared to [earlier explorations](https://cloudup.com/i8uQJGS9vuM), I think this works nicely to retain the dark focus state, while keeping the icons strong enough to not compete with the borders. 

Please head over to #14197 and give that a spin too. 👍

**Comment by afercia** [comment]
A more in depth discussion on the block breadcrumb is happening on #14095 and I'd agree it would be good to understand [what problem these breadcrumbs solve](https://github.com/WordPress/gutenberg/issues/) in the first place. If they're going to stay, I'd like to share a couple thoughts:

When exploring the "navigation / edit mode" in #3195 / #5694 at some point it was clear there was the need for a `<button>` element in each block to switch between the two modes. This exploration hasn't seen much progress in the last months but it still might be one of the things to try to improve the overall keyboard interaction with the blocks. There's a chance something like that might be implemented in the future and just wanted to mention the "breadcrumb" could be used as the button we need to switch mode. Just a consideration I wanted to share, nothing actionable here at the moment.

Personally, I've never fully understood why the breadcrumbs are so small :) The look like something unfinished, stuck between the attempt to communicate the block type and the will to not "disturb" users too much. Being so small, their white space is inconsistent with the one used across the UI and they feel a bit extraneous to the overall UI look and feel. Why not consider to make them bigger? Other applications use similar overlays and they're not afraid to temporarily hide other parts of the UI. For example, even if it's a slightly different case, here's what Squarespace does:

squarespace quote

**Comment by youknowriad** [comment]
I pushed a tweak to use a SASS variable to configure the size of the left border easily. I have a small preference for this approach personally. 

The darker border around the toolbars and the blocks seem odd to me as the inner border of the toolbar are lighter.

Anyway, happy to go with what you all feel is best.

**Comment by kjellr** [comment]
A small detail to note: 

If anyone has any idea why some of the measurements here are tiny portions of a pixel off on retina displays, please let me know: 

screen shot 2019-03-06 at 10 11 44 am

---

screen shot 2019-03-06 at 10 11 56 am

---

These don't seem like blockers by any means, but they're really annoying little bugs. Adjusting the position by 1px in either direction just makes things worse. 🤔

**Comment by chrisvanpatten** [comment]
@kjellr Probably something to do with browser subpixel rounding. This will sound nuts, but can you try moving a fraction of a pixel? Some browsers will actually obey subpixel values! It's weird but they do!

(e.g. `left: 1.1px` or whatever the values are.)

I think you can also work around some subpixel issues using `transform`s but that can get hairy and has performance benefits.

Also, I agree with @afercia. I would love to see an exploration that unifies the toolbar and the label, with something like this:

exploration

This simplifies and unifies the hover and selected states, reducing the amount of changing UI a bit.

You could also imagine this with a discreet "edit" button like @afercia suggested:

exploration-edit

**Comment by kjellr** [comment]
> This will sound nuts, but can you try moving a fraction of a pixel? Some browsers will actually obey subpixel values! It's weird but they do!

Yeah, I've been messing around with that a bit. It _almost_ works, but feels really messy. 😄 

My assumption is that if subpixel tweaks are necessary, there must be somewhere that's throwing things off exact pixels in the first place... I've been digging to find the root of it. Haven't found anything yet though... We tend to use exact pixels in the position calculations here, so I'm not sure where this is originating. 

Regarding the labels — that's an interesting thought. It's actually really similar to [the original design](https://github.com/WordPress/gutenberg/pull/5658) of these labels: 

37700066-6e9c98d8-2cea-11e8-9929-c1e8850ce800

There's actually some discussion around the breadcrumbs in https://github.com/WordPress/gutenberg/issues/14095#issuecomment-468778570, asking whether or not they're actually necessary at all. These breadcrumbs were added before blocks had permanent block indicator icons in their toolbars. It's interesting to see two divergent paths here. 😄

**Comment by afercia** [comment]
> These breadcrumbs were added before all blocks had permanent block indicator icons in their toolbars. It's interesting to see two divergent paths here.

Yep 🙂 The "permanent block indicator icons" are, well... icons, with all their limitations. :) Text is always better for accessibility. The "Edit" button would finally give a better sense to these breadcrumbs, if the "navigation - edit mode" will see some progress.

Re: the subpixels thing: I guess it's really a browser thing. I don't see it happening with Firefox or Safari (which is interesting). Also, if I change the _hover_ border to the same dark rgba color of the selected border, it happens also with the hover border (with Chrome).

**Comment by kjellr** [comment]
> Yep 🙂 The "permanent block indicator icons" are, well... icons, with all their limitations. :) Text is always better for accessibility. The "Edit" button would finally give a better sense to these breadcrumbs, if the "navigation - edit mode" will see some progress.

I'm just reading up now on navigation/edit mode thinking but in general, would it make sense to only show the breadcrumb when in navigation mode? When I look at this mockup for instance: 

53892942-3af53380-3ffb-11e9-951f-6f8fd405c6ec

I worry that the edit icon would be misleading on hover, as it'd seem to imply that the user needs to click on it to edit the block. When in fact, they can just click anywhere. But if it's helpful for someone who's using navigation mode, then we could show it there only. Does that seem right?

In any case though — to keep us on track: this PR is focused on reducing the hover borders, and providing greater contrast to the focus state. The current breadcrumb treatment in the PR is really just a relocation + color change to what we have now, so I think it would make sense for us to explore breadcrumb-specific enhancements separately.

**Comment by mapk** [comment]
> this PR is focused on reducing the hover borders, and providing greater contrast to the focus state. 

Good call, Kjell. I think more conversation around labels could separate out to their own issue so that this can stay focused.

I'm very fond of this heavier left border version. The way it ties the toolbar and block together is slick. I also feel it's lighter UI doesn't give me the feeling of being too "blocky." 

How does everyone feel about this implementation within the context of a full site editor? Do you foresee any problems there? Just trying to future-proof this design choice if at all possible.

A few more comments regarding the columns block with this design.

1. The column block, without a full border on hover becomes more difficult to select. This might be resolved with a few other PRs out there right now though.
2. The inner block's hover state is blocked from view by the toolbar of the column block.

col

**Comment by afercia** [comment]
> I worry that the edit icon would be misleading on hover

I don't want to derail conversation too much. Briefly: in the navigation/edit mode scenario, the edit icon may not be strictly necessary: mouse/touch users already click on the block to edit. Keyboard users need a button, that could be the breadcrumb itself, even without the edit icon.

**Comment by jasmussen** [comment]
Kjell, be sure to test whether that subpixel issue is Safari only. If it is, I would bet that it's related to the rendering mode applied to the block. I.e. it may be enough to apply something like `will-change: transform;` to the hover label.

**Comment by kjellr** [comment]
> How does everyone feel about this implementation within the context of a full site editor? Do you foresee any problems there? Just trying to future-proof this design choice if at all possible.

This is sort of hard to gauge without having a clearer picture of what this would look like. I think the closest meaningful indicator we have at the moment would be how this works with nested blocks, as I can imagine a full-site editor would involve many of those. More on that below.

> A few more comments regarding the columns block with this design.

> 1. The column block, without a full border on hover becomes more difficult to select. This might be resolved with a few other PRs out there right now though? 
> 2. The inner block's hover state is blocked from view by the toolbar of the column block.

1. Yeah, selecting columns in general needs to be sorted out, so I think that can be solved via #9628
2. Yeah. This is not ideal. I can maybe adjust some z-indexes there? I'll take a look. It looks like the lack of padding in there may also be problematic:

screen shot 2019-03-07 at 9 53 41 am

---

In any case, in order to get a better sense of how this might work with nested blocks, I made a figma prototype of the concept we've been discussing in #9628. Give it a spin, and definitely compare it to the approach in #14197 (I created a similar prototype to compare that approach too 🙂)

[🖥 **Figma Prototype**](https://www.figma.com/proto/Jg0F7WNWL9mAecweJZrKJe6R/Block-Borders?node-id=18%3A1501&viewport=-551%2C-276%2C0.25&scaling=min-zoom)

**Comment by jasmussen** [comment]
You continue to amaze, Kjell. What you're doing here with _two_ separate PRs that approach this difficult tasks, wade through so much feedback, yet still manage to create wonderful prototypes, it's just stellar. Well done.

With regards to the _visuals_, while I personally have a preference for #14197 with the singular color and uniform border thickness, it seems relatively clear that the overall preference from the rest of this group is towards the version that has a thicker left border. My preference stems from the idea that simplicity is fewer details, and the thicker left border feels like a new ingredient to the mix. But, since my personal opinion is irrelevant 😅 (thank goodness!), if the visual consensus is for the thick-border version, I will learn to love it! 

Specifically with regards to the prototype, it strongly feels like:

a) hover style is not as important as we may feel it is at the moment
b) the hover label as well, doesn't feel that important

but rather, that what matters is that when you are in a situation with lots of nesting, it really does help a lot to have the parent block highlighted in some way. Whether the thicker left border, vs. the same-color-all-around 1px border is better, that's unclear, since it's only the _selected_ and _hovered_ blocks that receive this treatment, not adjacent or parent blocks. 

We've tried for a long time to crack the nut that is edit/navigation mode, where I still feel like https://github.com/WordPress/gutenberg/issues/5694#issuecomment-413798567 may hold the ingredients to make it work. However the actual _hover_ style is not a blocker for moving forward with that, nor even is removing any hover labels. Because navigation mode is by definition very keyboard heavy, it's probably best to tackle that separately, and not block hover changes on that.

**Comment by kjellr** [comment]
Hey friends! After some discussion with @jasmussen over in the other thread (https://github.com/WordPress/gutenberg/pull/14197#issuecomment-471544154), I've added a slightly darker border to selected blocks in this PR. This change: 

1. Balances out the left border a bit, to make things a bit less left-heavy. 
2. Improves contrast, which is always a plus. 
3. Allows for use the old lighter color for complimentary borders if we decide to later (parent/child block border outlines, etc). 
4. Is subtle enough that the toolbar icons are still visibly darker than the border. 
5. Is a bit of a bridge between this direction and the one in #14197. With this in, and with the general leaning towards this direction, I think we can safely close PR issue in favor of this one. 

54128008-76b84080-43e1-11e9-84e1-decaf2e74b0b

I've also tested & cleaned up the appearance of this change when themes with a dark background: 

Screen Shot 2019-03-11 at 11 18 59 AM

Please give this a test, and let me know if: 

1. This darker border works.
2. You find any bugs. 

Thank you!

**Comment by chrisvanpatten** [comment]
That bottom pixel of space below the thick border is super painful… I suppose that might be a rounding issue?

Screen Shot 2019-03-11 at 11 24 05 AM

**Comment by kjellr** [comment]
> That bottom pixel of space below the thick border is super painful (okay I'm being a smidge dramatic)

Yeah, I know about it. 😭 I haven't found an elegant solution, unfortunately. The issue is that the block outline lies outside of the bounds of the block. So a left box shadow (even one that's externally placed) will not extend to the top and bottom to cover that empty spot. 

So far, ideas to fix this include: 

1. Change the entire block outline to be a `border` instead of an `outline`, specify that the bottom should have _no_ border, and then add a fake bottom border as an inset `box-shadow`. 
2. Change the entire block outline to be a `border` instead of an `outline`, specify that the bottom should have _no_ border, and then implement that bottom border as a second pseudo-element, to line up with the left border. 

Both of those seem like total hacks though 😕 I'd love to see if anyone has other ideas... hopefully I missed something obvious.

**Comment by kjellr** [comment]
I pushed a few updates. First, I was able to solve the border issue @chrisvanpatten noted, by switching to `border` instead of `outline` for our block borders. This appeared to have (_almost_) no ill effects on anything: 

Screen Shot 2019-03-11 at 2 26 21 PM

A couple things that could use some 👀:

---

**Windows High Contrast Mode**

The only thing that _did_ get worse when I switched to borders was the way this looks in Windows High Contrast mode. By default, there was no left border at all. I tried adding a `border-left: 1px solid transparent`, but that gave me this pixel error everywhere:

Screen Shot 2019-03-11 at 2 32 10 PM

So instead, I added a whole `outline` to the block: 

Screen Shot 2019-03-11 at 2 27 59 PM

This is less than ideal, because it means there's a double border. 😕 I'd love some thoughts on whether this is bad or not. 

---

**Opaque vs. Semi-transparent Left Borders**

Also, in the latest updates, I switched the thick left border to be fully opaque. When it was transparent, we'd frequently get some [odd overlaps](https://cloudup.com/cEPGKH0VJ0q). This fixes that. Since the color is really dark anyway, it seems to work just fine against just about any light-ish background I throw behind it: 

Screen Shot 2019-03-11 at 2 37 34 PM
Screen Shot 2019-03-11 at 2 37 49 PM

As soon as it does start getting weird, the theme should probably end up calling itself a dark theme to get the alternate controls in general:

Screen Shot 2019-03-11 at 2 38 10 PM

Does this seem fine to everyone?

Hover states have been fully opaque since the start of this PR, but if anything I think those _could_ benefit by switching to the semi-transparent version: 

_Current_
Screen Shot 2019-03-11 at 2 50 38 PM

_Alternative, semi-transparent hover state:_ 
Screen Shot 2019-03-11 at 2 48 26 PM

Unfortunately, that'd mean that unless the background were white, the breadcrumb would not match the color of the left border (since we need to maintain readability there). 😕 Not sure how I feel about that. 

---

The good news is: these are all fairly minor details. So I feel like we're getting closer. 🙂

**Comment by jasmussen** [comment]
GIF of the latest state:

GIF

Key take-away, the slightly darker overall border does a great deal to _balance_ things out. It still feels slightly left-leaning, but not in what I personally felt was a slightly tricky way. Nice work.

I still personally have a preference for the dark left hover border, i.e. one that doesn't chagne color from hover to select. But since that's just a personal preference, I will completely defer to you on this one. 

Nice work checking high contrast mode too! It feels like it's okay to be "more opinionated" there, as it's an intentional mode people choose and therefore okay to turn up the contrast to 11. Seems like once the editor-style details are nailed down, this is shippable! @alexislloyd any thoughts on the final state of this?

**Comment by kjellr** [comment]
Minor update: pushed a small fix for the [nested block issue noted above](
https://github.com/WordPress/gutenberg/pull/14145#issuecomment-470560248). I just adjusted the z-index of child block breadcrumbs so they float above the toolbar. 

_Before_
Screen Shot 2019-03-12 at 8 49 19 AM

_After_
Screen Shot 2019-03-12 at 8 48 53 AM

**Comment by SchneiderSam** [comment]
This PR is so awesome. I can hardly wait. You are so great!!!

**Comment by kjellr** [comment]
I pushed another round of minor updates this afternoon: 

---

This PR now hides the thick border of the block toolbar when Top Toolbar mode is active: 

Screen Shot 2019-03-12 at 2 45 51 PM

---

I've extended the left border to the Permalink area of the post title. This mimics the block toolbar's border behavior, and feels less disjointed than what was there before.

Screen Shot 2019-03-12 at 2 44 24 PM

---

I fixed some tiny bugs on mobile screens. Before, the bottom border of the block toolbar was the old, lighter color. Now it's fixed: 

Screen Shot 2019-03-12 at 2 59 27 PM

In addition, floated toolbars on mobile used to suffer from the same [missed-pixel issue noted here](https://github.com/WordPress/gutenberg/pull/14145#issuecomment-471585430). That's sorted out now too:

Screen Shot 2019-03-12 at 2 49 26 PM

**Comment by mapk** [comment]
I ran through this today and it felt really good. The darker border on the selected blocks wasn't bad at all. Thanks for your persistence in making this a reality, @kjellr! Also @jasmussen thanks for your insight and feedback!

I tested all the core blocks (except the embeds), and on FF and Safari. 

🚢

**Comment by mapk** [comment]
I just caught one little thing in the Classic block. The left border on that icon looks pretty heavy.

classic

**Comment by kjellr** [comment]
Thanks, @mapk! I've cleaned that up, as well as similar issue with the other special-case blocks: reusable + warnings: 

special-case-blocks

**Comment by jasmussen** [comment]
This is really impressive work. Took it for a solid spin.

Screenshot 2019-03-15 at 09 05 13

That's a clever way of handling the left borders on fullwide.

The permalink box is a bit offset. But you _don't_ have to fix this, as the permalink box is going to be revisited separately. But figured I'd mention it regardless:

Screenshot 2019-03-15 at 09 05 37

The following is not a regression in this PR, master and this branch are the same, but CC: @jorgefilipecosta  — aren't the block outlines supposed to be inverted if the background dim is dark enough inside the Cover block? This branch:

Screenshot 2019-03-15 at 09 06 10

Master:

Screenshot 2019-03-15 at 09 11 06

On that note, in general the outlines seem to account decently for dark editor styles:

Screenshot 2019-03-15 at 09 07 27

Except for the hover state:

Screenshot 2019-03-15 at 09 07 36

I don't think we account for the hover state in _master_ so this is technically not a regression. Can you think of a good way to make that hover state work better for inverted styles? If not, then it's not a blocker.

There's a teensy bit of funkiness going on at the mobile breakpoint. Here's TwentyNineteen:

Screenshot 2019-03-15 at 09 08 02

Here's the default style:

Screenshot 2019-03-15 at 09 08 40

Those mobile styles are the only ones that feel like they could use a tiny touch-up. But other than that, 👍 👍 from me, and well done!

**Review by jasmussen — APPROVED** [review]
I left some thoughts i https://github.com/WordPress/gutenberg/pull/14145#issuecomment-473196181. But they are so minor (small mobile issue in twentynineteen that was probably already present, and small improvement to hover states in dark themes), that I wanted to approve this regardless.

It's a welcome refinement, it's a big PR, and in order to not hold off improvements in the name of perfection, I think it can be shipped with subsequent small followup PRs.

**Comment by kjellr** [comment]
> The permalink box is a bit offset. But you don't have to fix this, as the permalink box is going to be revisited separately. But figured I'd mention it regardless:

Yeah, I noticed that in Master too. 😕 I haven't looked into it yet, but I wonder if it has to do with Twenty Nineteen's editor styles. We can fix separately.

Screen Shot 2019-03-15 at 8 20 47 AM

> I don't think we account for the hover state in master so this is technically not a regression. Can you think of a good way to make that hover state work better for inverted styles? If not, then it's not a blocker.

I've pushed c3b6c3b, which uses an alternate dark color for the hover + breadcrumb for dark themes: 

dark-theme

> There's a teensy bit of funkiness going on at the mobile breakpoint. Here's TwentyNineteen

Ah, right. Usually, the block borders extend to the screen edges on mobile. That doesn't happen on Twenty Nineteen (If I recall correctly, it's because Twenty Nineteen includes some margins around the editing area or something like that). Anyway, thanks for the heads up. This should be fixed in 9b5325f:

Screen Shot 2019-03-15 at 8 40 17 AM

**Comment by jasmussen** [comment]
🆒

**Comment by youknowriad** [comment]
I just noticed that the post title hover and selected styles are not great in "Spotlight Mode"

**Comment by kjellr** [comment]
@youknowriad can you share what you're seeing? I don't think there are supposed to be any hover style there in Spotlight Mode, and the select style should be just like all the other blocks in that mode: 

hover-title

**Comment by youknowriad** [comment]
Oh I have blue outlines, maybe it's related to the branch I was in. Sorry for the ping, I'll check deeper.

**Comment by irishetcher** [comment]
It's a bit better, but still not user friendly. On testing I accidentally dropped a Kadence row inside a Kadence row and was wondering why I couldn't make the inner one go fullscreen. It wasn't evident what had happened until I used the document drop down to see where I had gone. Then when I tried to move the row block down out of the outer row it stubbornly wouldn't nudge down the theme footer and ended up floating around after my cursor (with no click down on my mouse button). To resolve I had to scrap the rows and start over.

File: packages/block-editor/src/components/block-list/style.scss
Comment (aduth) [inline packages/block-editor/src/components/block-list/style.scss]: Is it on the left, or on the right?

There are still existing styles which assume right-placement for the wide/full hover label:

https://github.com/WordPress/gutenberg/blob/e209cfb77469b0e022198f75ab1fd641f98b229a/packages/block-editor/src/components/block-list/style.scss#L427-L430

I'm supposing that left placement is reasonable only in part due to the regression of #14817 

Fixing #14817 introduces an undesirable overlap:

image

Comment (aduth) [reply packages/block-editor/src/components/block-list/style.scss]: > Fixing #14817 introduces an undesirable overlap:

Follow-up: #15022 (proposes moving to the right)

**Pull request merged**
